modules/internal/ChapelLocale.chpl

```diff
@@ -286,25 +410,25 @@ module ChapelLocale {
     pragma "no doc"
     proc defaultMemory() : locale {
       HaltWrappers.pureVirtualMethodHalt();
-      return this;
+      return new locale(this);
```

Comment:
Probably an arbitrary choice here, but I wonder if this should return `nilLocale` (or actually I wonder if we can just remove these returns now, it wouldn't surprise me if the compiler has gotten better at not requiring returns for functions that halt)

Reply:
This worked except only one case where the `override` method was recognized as returning `void` even though the return type was explicitly `locale`. https://github.com/chapel-lang/chapel/pull/15149/files#diff-d3f589f00f743ac7c8fe7542db13be39R238.

I return `defaultLocale` there, will probably open an bug issue about the problem.